### DISCUSSION TITLE
Should "Prebuild triggers": "On configuration change" include Dockerfile changes?

### DISCUSSION THREAD

**Discussion by aristocrates** [community/community#40737] [Codespaces] [answered]
### Select Topic Area

Question

### Body

I have a repository with `.devcontainer/devcontainer.json` specifying `"dockerFile": "Dockerfile"`, and a prebuild configuration just for branch `main` with "Prebuild triggers" set to "On configuration change".

When I pushed a change to `.devcontainer/Dockerfile`, no prebuild was triggered.

From the description of "On configuration change" and the docs:

> On configuration change
>
> Your codespace will prebuild when a change is detected on the devcontainer.json and associated configuration files

Screenshot:
<img width="797" alt="Screenshot 2022-12-03 at 9 28 55 PM" src="https://user-images.githubusercontent.com/4792145/205476130-65289e6f-884e-47b8-9e88-0cca68475eda.png">

What files or file types does "associated configuration files" refer to? Should the specified `dockerFile` count as an associated configuration file? Is there a way to ensure that changes to `.devcontainer/Dockerfile` trigger a prebuild even if there are no changes to `.devcontainer/devcontainer.json`, without building on every push?

**Comment by eljog** [comment, marked as answer]
Hi @aristocrates thank you for reporting this issue, and sorry to hear that you had a trouble using the service.

Upon investigation we found that there is a bug that causes changes in certain configs are not accurately detected for triggering a prebuild. Our engineering team is looking into it.
We are also working on improving the documentation to explain the supported scenarios for `on configuration change` better.
